Improved visuals of main menu and some UX features

w050asg opened the pull request:
Altered the look of the main menu by adding a title and using different colors to improve the look. Got strings to display to allow the user to determine what each button does. UX improvements like opening in fullscreen and exiting on escape to make it easier for the user.

w050asg committed:
Improved visuals of main menu and some UX features

Altered the look of the main menu by adding a title and using different colors to improve the look. Got strings to display to allow the user to determine what each button does. UX improvements like opening in fullscreen and exiting on escape to make it easier for the user.

NicholaiThe3rd commented:
Looks good so far, still attempting to find a way to implement these as buttons but I appreciate how clean it looks now. Possibly make the last section as an "Exit" button instead since we currently don't have any plans to add an options menu at this point. 

If some more comments to help the documentation would be possible I would appreciate it.

w050asg committed:
Button Logic and launching Craft

Added the mouseclick handler so we could determine where the user clicked and if it was a button. Added implementation to play game so the user can launch craft and changed options to exit game since I don't think options will be needed. Also added some comments for clarity

w050asg committed:
Revert "Button Logic and launching Craft"

This reverts commit 714ff9c04124a1f77b904e60438fc275ffe4ae7c.

w050asg committed:
Merge branch 'master' into Austin-Title

jrichards1999 commented on src/main.c:
```diff
@@ -2624,33 +2629,61 @@ void render(void){
+    glLoadIdentity();
+    glColor3f(1.0f,1.0f,1.0f);
+    glRasterPos2f(0.45, 0.76);
+    glutBitmapString(GLUT_BITMAP_TIMES_ROMAN_24, "Play Game");
```
This looks like a pretty convenient way to display text, wish I had used it for my project!

jrichards1999 replied:
This explains why the menu pops up at the start of main.  Sorry for missing that earlier!

jrichards1999 commented on src/main.c:
```diff
@@ -2624,33 +2629,61 @@ void render(void){
 glutInitDisplayMode (GLUT_SINGLE | GLUT_RGB);
-glutInitWindowSize (800, 600);
-glutInitWindowPosition (100, 100);
+glutInitWindowSize (glutGet(GLUT_SCREEN_WIDTH), glutGet(GLUT_SCREEN_HEIGHT));
```
Good use of descriptive variables here to make the code more readable.

jrichards1999 reviewed (commented):
Great work, Austin.  The changes made improve code readability, but could still use some extra comments to help explain what is happening in the renderButton function.

NicholaiThe3rd reviewed (commented):
Looks good so far, still attempting to find a way to implement these as buttons but I appreciate how clean it looks now. Possibly make the last section as an "Exit" button instead since we currently don't have any plans to add an options menu at this point.

If some more comments to help the documentation would be possible I would appreciate it.
